docs/usage/interpreter/overview.md

```diff
@@ -123,3 +123,11 @@ Snippet of code (language of interpreter) that executes after initialization of
 <img src="{{BASE_PATH}}/assets/themes/zeppelin/img/screenshots/interpreter_precode.png" width="800px">
 
  
+## Interpreter Lifecycle Management
+
+Before 0.8.0, Zeppelin don't have lifecycle management on interpreter. User have to shutdown interpreters explicitly via UI. Starting from 0.8.0, Zeppelin provides a new interface
+`LifecycleManager` to control the lifecycle of interpreters. For now, there're 2 implementations: `NullLifecycleManager` and `TimeoutLifecycleManager` which is default. 
+
+`NullLifecycleManager` will do nothing,
+user need to control the lifecycle of interpreter by themselves as before. `TimeoutLifecycleManager` will shutdown interpreters after interpreter idle for a while. By default, the idle threshold is 1 hour.
```

Comment:
how is this idleness defined?

Reply:
It is `zeppelin.interpreter.lifecyclemanager.timeout.threshold` mentioned below

Reply:
what I mean is, how do we know the interpreter is idle? for example SparkRepl could be running a big job and for us the paragraph is "running"

Reply:
If the paragraph is running, the interpreter is not idle. Because zeppelin server will poll job status from interpreter process, and zeppelin will update the lastActivityTime.